backend/src/graphql/resolver/util/communities.ts

```diff
@@ -63,3 +69,55 @@ export async function getCommunityByUuid(communityUuid: string): Promise<DbCommu
     where: [{ communityUuid }],
   })
 }
+
+/**
+ * Simulate RIGHT Join between Communities and Federated Communities
+ * select *
+ * Community as c
+ * RIGHT JOIN FederatedCommunity as f
+ * ON(c.public_key = f.public_key)
+ * Typeorm don't has right joins
+ * @returns
+ */
+export async function getAllCommunities({
+  pageSize = 25,
+  currentPage = 1,
+}: Paginated): Promise<DbCommunity[]> {
+  const connection = await Connection.getInstance()
+  if (!connection) {
+    throw new LogError('Cannot connect to db')
+  }
+  // foreign: 'ASC',
+  // createdAt: 'DESC',
+  // lastAnnouncedAt: 'DESC',
+  const result = await connection
+    .getRepository(DbFederatedCommunity)
+    .createQueryBuilder('federatedCommunity')
+    .leftJoinAndSelect('federatedCommunity.community', 'community')
+    .orderBy('federatedCommunity.foreign', 'ASC')
+    .addOrderBy('federatedCommunity.createdAt', 'DESC')
+    .addOrderBy('federatedCommunity.lastAnnouncedAt', 'DESC')
+    .skip((currentPage - 1) * pageSize * 3)
+    .take(pageSize * 3)
+    .getManyAndCount()
+  const communityMap = new Map<string, DbCommunity>()
+  result[0].forEach((value: DbFederatedCommunity) => {
+    const publicKeyHex = value.publicKey.toString('hex')
+    if (!communityMap.has(value.publicKey.toString('hex'))) {
+      let community: DbCommunity = DbCommunity.create()
+      if (value.community) {
+        community = value.community
+      }
+      if (!community.federatedCommunities) {
+        community.federatedCommunities = []
+      }
+      communityMap.set(publicKeyHex, community)
+    }
+    const community = communityMap.get(publicKeyHex)
+    if (!community?.federatedCommunities) {
+      throw new LogError('missing community after set it into map', publicKeyHex)
+    }
+    community.federatedCommunities.push(value)
+  })
+  return Array.from(communityMap.values())
+}
```

Comment:
Why simulating a RightJoin here in TypeScript when the database could select all the data like you need for you graphQL-model?

Imaging the process how both tables are filled:
1. Communities: the homeCom-Entry is created as soon the DHT starts
2. federatedCommunity: the homeCom-Entry is created as soon the DHT starts
3. federatedCommunity: the DHT writes foreignCom-Entries during federation
4. federatedCommunity: the validateCommunity-Process exchange info per direct graphql-communication and mark the foreign federatedCom as verified
5. Communities: the validateCommunity-Process starts for a verified federatedCom the authenticateCommunity-Process and creates a foreignCom-Entry in communities table
6. Communities: the authenticateCommunity-Process updates the foreignCom-Entry after successfull exchange of both communityUuids

Your master-table will be the `federatedCommunities` and for each different `publicKey` - means you need `group by publicKey` - an associated entry in `communities` could exists

Reply:
My master table is federatedCommunities
```ts
.getRepository(DbFederatedCommunity)
```
I use communityMap for sorting by publicKey

Reply:
- I retrieve federatedCommunity entries together with community entry if exist
- for each federatedCommunity:
  - check if community already exist in communityMap by public key
  - if not create empty Community entity and insert into map 
  - push federatedCommunity into federatedCommunities array from community

Finally:
- I have a Community Entity for each different public key in federated_communities
- the contain a array with at least on federatedCommunity
- if no community entry with this public key exist in db, the community entity has only federatedCommunities
- that's reason for empty checks and copy over values from federatedCommunity in model/Community